### PR TITLE
Set up sbt-ci-release.

### PR DESCRIPTION
**Pull request by sjrd**
Following the instructions of https://github.com/sbt/sbt-ci-release
I have already set up the secrets on the repo.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+name: Release
+on:
+  push:
+    branches: [main]
+    tags: ["*"]
```
Comment (julienrf) [inline .github/workflows/release.yml:5]: This will publish snapshots on every push to `main`, is that what you want?

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        with:
+          fetch-depth: 0
+      - name: Set up JDK 8
+        uses: actions/setup-java@v3
+        with:
+          distribution: 'adopt'
+          java-version: 8
+          cache: 'sbt'
+      - name: npm install
+        run: npm install
```
Comment (julienrf) [inline .github/workflows/release.yml:20]: Why is this needed?

**Review by julienrf — COMMENTED** [review]
Eventually, it would be great to also set `versionScheme` and set up bincompat checks (maybe use sbt-version-policy?)

File: .github/workflows/release.yml (hunk 1 above)
Comment (sjrd) [reply .github/workflows/release.yml:5]: Yes. We should have early feedback on whether the publishing doesn't publish "too much", at the very least.

File: .github/workflows/release.yml (hunk 2 above)
Comment (sjrd) [reply .github/workflows/release.yml:20]: We also use in the `ci.yml`, so I put it here for consistency of the environment.

But in theory it shouldn't be needed. Do you prefer if I remove it?

Comment (julienrf) [reply .github/workflows/release.yml:20]: I think it is better to omit it in this case (to avoid unnecessary computations)

Comment (sjrd) [reply .github/workflows/release.yml]: OK, I removed it.

**Review by julienrf — APPROVED** [review]
